File: Android_Zone_Test/src/com/example/mylib_test/activity/pop_dialog/Dialog_Pop_Adapter_MainActivity.java
package com.example.mylib_test.activity.pop_dialog;

import and.log.ToastUtils;
import view.DialogCustemZone;

import com.example.mylib_test.R;
import com.example.mylib_test.activity.pop_dialog.pop.Pop_Bottom;
import com.example.mylib_test.activity.pop_dialog.pop.Pop_Photo;

import android.app.Activity;
import android.app.AlertDialog.Builder;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.text.Spannable;
import android.text.SpannableString;
import android.text.style.AbsoluteSizeSpan;
import android.text.style.BackgroundColorSpan;
import android.text.style.ForegroundColorSpan;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.Toast;

public class Dialog_Pop_Adapter_MainActivity extends Activity implements OnClickListener{

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.a_pop_dialog_adapter);
	}
	@Override
	public void onClick(View v) {
		switch (v.getId()) {
		case R.id.pop:
//			PopCus pc=new PopCus(this, R.layout.poptest);
//			pc.showPop();
			Pop_Photo pop=new Pop_Photo(this, R.id.flowLayoutZone1);
			pop.show();
			break;
		case R.id.pop_bottom:
//			PopCus pc=new PopCus(this, R.layout.poptest);
//			pc.showPop();
			Pop_Bottom pop_bottom=new Pop_Bottom(Dialog_Pop_Adapter_MainActivity.this,R.id.pop_bottom);
			pop_bottom.show();
			break;
		case R.id.dialog:
			//dialog自定义测试
			DialogCustemZone dcz=new DialogCustemZone(this) {
				
				@Override
				public void notSure() {
					toToast("这个is not OK!");
				}
				
				@Override
				public void isSure() {
					toToast("这个is OK,流逼！");
				}

				@Override
				public void addSetProperty(Builder db) {
					db.setIcon(R.drawable.ic_launcher);
				}
				
			};
			break;
		case R.id.textGaoLiang:
			//点击文字高粱等效果
			Button bt = (Button) findViewById(R.id.textGaoLiang);
			Spannable span = new SpannableString("I am what are you doing，fucking!");
			span.setSpan(new AbsoluteSizeSpan(20), 1, 16,Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);//字体大小
			span.setSpan(new ForegroundColorSpan(Color.RED), 5, 16,Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);//前景色
			span.setSpan(new BackgroundColorSpan(Color.YELLOW), 11, 16,Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);//背景色
			bt.setText(span);
//			bt.setText(Html.fromHtml(  "<font color=#E61A6B>红色代码</font> "+
//			"<I><font color=#1111EE>蓝色斜体代码</font></I>"+"<u><i><font color=#1111EE>蓝色斜体加粗体下划线代码</font></i></u>"));
			break;

		default:
			break;
		}
	}
	public void toToast(String str) {
		ToastUtils.showShort(this, str);
	}
}

File: Android_Zone_Test/src/com/example/mylib_test/activity/pop_dialog/Dialog_Pop_Adapter_MainActivity.kt
package com.example.mylib_test.activity.pop_dialog

import com.example.mylib_test.activity.pop_dialog.dialog.ZDialog
import com.zone.lib.utils.activity_fragment_ui.ToastUtils

import view.DialogCustemZone

import com.zone.view.FlowLayout

import com.example.mylib_test.R
import com.example.mylib_test.activity.pop_dialog.pop.Pop_Bottom
import com.example.mylib_test.activity.pop_dialog.pop.Pop_Photo
import android.app.AlertDialog.Builder
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.view.View
import android.widget.Button
import com.example.mylib_test.LogApp
import com.example.mylib_test.activity.pop_dialog.pop.PopFullScrenn
import com.zone.lib.ZLogger
import com.zone.lib.base.controller.activity.BaseFeatureActivity

class Dialog_Pop_Adapter_MainActivity : BaseFeatureActivity() {

    override fun setContentView() {
        setContentView(R.layout.a_pop_dialog_adapter)
    }

    override fun initData() {
    }

    override fun setListener() {
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.pop -> Pop_Photo(this, R.id.flowLayoutZone1, R.id.pop).show()
            R.id.popFull -> PopFullScrenn(this).show()
            R.id.pop_bottom -> Pop_Bottom(this, R.id.pop_bottom).show()
            R.id.tag -> tabBlue()
            //启动FxService
            R.id.bt_floatView -> startService(Intent(this, FxService::class.java))
            R.id.bt_floatViewClose -> floatViewClose()
            R.id.custom_dialog -> customDialog()
            R.id.dialogFullScreen -> ZDialog(this).show()
            R.id.bt_floatNotification -> startService(Intent(this, NoticationService::class.java))
            else -> {
            }
        }
    }

    private fun customDialog() {
        object : DialogCustemZone(this) {

            override fun notSure() {
                toToast("这个is not OK!")
            }

            override fun isSure() {
                toToast("这个is OK,流逼！")
            }

            override fun addSetProperty(db: Builder) {
                db.setIcon(R.drawable.ic_launcher)
            }

        }.show()
    }

    private fun tabBlue() {
        val fz = findViewById<View>(R.id.flowLayoutZone1) as FlowLayout
        val bt2 = fz.findViewWithTag<View>("blue") as Button
        bt2.setBackgroundColor(Color.BLUE)
    }

    private fun floatViewClose() {
        //uninstallApp("com.phicomm.hu");
        val intent2 = Intent(this, FxService::class.java)
        //终止FxService
        stopService(intent2)
    }

    fun toToast(str: String) {
        ToastUtils.showShort(this, str)
    }


    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        LogApp.d("R.id.pop  onPostCreate:${findViewById<View>(R.id.pop).height}")
    }

    override fun onPostCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onPostCreate(savedInstanceState, persistentState)
        LogApp.d("R.id.pop  onPostCreate2:${findViewById<View>(R.id.pop).height}")
    }
    override fun onPostResume() {
        super.onPostResume()
        LogApp.d("R.id.pop  onPostResume:${findViewById<View>(R.id.pop).height}")
    }

    override fun onStart() {
        super.onStart()
        LogApp.d("R.id.pop  onStart:${findViewById<View>(R.id.pop).height}")
    }
    override fun onResume() {
        super.onResume()
        LogApp.d("R.id.pop  onResume:${findViewById<View>(R.id.pop).height}")
    }
}
